vignettes/more-recipes.Rmd

```diff
@@ -0,0 +1,277 @@
+---
+title: Apply other steps and models from recipes and parsnip packages
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Apply other steps and models from recipes and parsnip packages}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+The `epipredict` package utilizes the `tidymodels` framework, namely 
+[`recipes`](https://recipes.tidymodels.org/) for 
+[dplyr](https://dplyr.tidyverse.org/)-like pipeable sequences 
+of feature engineering and [`parsnip`](https://parsnip.tidymodels.org/) for a 
+unified interface to a range of models. 
+
+`epipredict` has additional customized feature engineering and preprocessing steps 
+tailored to fit the needs of epidemiology forecasting, such as `step_epi_shift()`, 
+`step_population_scaling()`, `step_epi_naomit()`. They can be used along with 
+steps from the `recipes` package for more feature engineering. 
+
+In this vignette, we will demonstrate how to weave `recipes` steps and `parsnip`
+models into the `epipredict` workflow. 
+
+First, let's load the necessary packages:
+
+```{r, warning=FALSE, message =FALSE}
+library(epipredict)
+library(epiprocess)
+library(recipes)
+library(parsnip)
+library(workflows)
+library(poissonreg)
+```
+
+We will use the built-in dataset `case_death_rate_subset` and apply feature 
+engineering steps and fit a few models from `parsnip` to predict future death 
+rates by states 7 days ahead from Dec 31, 2021.
+
+```{r}
+x <- case_death_rate_subset
+x$geo_value <- as.factor(x$geo_value)
+tail(x)
+```
+There are `r dim(x)[1]` rows and `r dim(x)[2]` columns in the data, with dates
+ranging from `r min(x$time_value)` to `r max(x$time_value)` in 
+`r n_distinct(x$geo_value)` states in the U.S.
+
+# Preprocessing and Feature Engineering
+
+In this section, we focus on showing how `step_*()` functions from the `recipes`
+package can also be applied in our `epipredict` workflows. 
+
+First, create a recipe object from the original data and then specify the 
+preprocessing steps. Steps are then sequentially added.
+
+`recipes` package also allows [manual alterations of roles](https://recipes.tidymodels.org/reference/roles.html) 
+in bulk. There are a couple handy functions that can be used together to help us 
+manipulate variable roles easily. 
+
+`update_role()` alters an existing role in the recipe or assigns an initial role 
+to variables that do not yet have a declared role.
+
+`add_role()` adds an additional role to variables that already have a role in 
```

Comment:
do this to geo_type and then convert to factor to use for dummy variables.

Reply:
use this along with add_role() on time_value and dummy geo_values to add in the model as covariates.

Show how to use formula in epi_recipe() as well.